react-app/src/components/main/ChooseDate.js

```diff
@@ -18,79 +20,50 @@ const style = {
     justifyContent: 'center',
     margin: '20px 0'
   }
-};
+}));
 
-export default class ChooseDate extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      redirect: false,
-      room: undefined,
-      date: new Date()
-    };
+function ChooseDate({
+  location: {
+    state: { room }
   }
+}) {
+  const classes = useStyles();
+  const [redirect, doRedirect] = useState(false);
+  const [date, setDate] = useState(moment(new Date()));
```

Comment:
calling `moment()` itself creates a new moment object according to the current time